File: scripts/_diagnose_live.py
import sqlite3
from datetime import datetime, timezone

db = sqlite3.connect('packages/db/prisma/dev.db')
db.row_factory = sqlite3.Row

def fmt_ts_ms(ts_str):
    try:
        # ISO string from prisma
        return datetime.fromisoformat(ts_str.replace('Z', '+00:00')).strftime('%H:%M:%S')
    except Exception:
        return str(ts_str)

# 1. What profiles exist?
print('=== PROFILES IN DB (CopyIntent) ===')
for r in db.execute("SELECT DISTINCT profileId, COUNT(*) as cnt FROM CopyIntent GROUP BY profileId"):
    print(f'  {r["profileId"]}: {r["cnt"]} intents')

# 2. Live profile intents
print()
print('=== LIVE PROFILE - Last 30 intents ===')
rows = db.execute("""SELECT ts, side, status, reason, mode FROM CopyIntent
                     WHERE profileId='live' ORDER BY ts DESC LIMIT 30""").fetchall()
if not rows:
    print('  NO INTENTS FOUND for profileId=live')
else:
    for r in rows:
        print(f'  {fmt_ts_ms(r["ts"])} {r["side"]:4} {r["status"]:10} reason={r["reason"] or "-":30} mode={r["mode"]}')

# 3. Alerts for live profile
print()
print('=== LIVE PROFILE - All alerts ===')
rows = db.execute("""SELECT ts, severity, code, message FROM Alert
                     WHERE profileId='live' ORDER BY ts DESC LIMIT 30""").fetchall()
if not rows:
    print('  No alerts for profileId=live')
else:
    for r in rows:
        print(f'  [{fmt_ts_ms(r["ts"])}] {r["severity"]:7} {r["code"]}: {r["message"][:100]}')

# 4. RuntimeMetrics for live
print()
print('=== LIVE PROFILE - RuntimeMetrics ===')
rows = db.execute("SELECT key, value, updatedAt FROM RuntimeMetric WHERE profileId='live'").fetchall()
if not rows:
    print('  No metrics for profileId=live')
else:
    for r in rows:
        print(f'  {r["key"]}: {r["value"]}')

# 5. Leader events for live
print()
print('=== LIVE PROFILE - Leader events (last 20) ===')
rows = db.execute("""SELECT ts, side, price, usdcSize, tokenId FROM LeaderEvent
                     WHERE profileId='live' ORDER BY ts DESC LIMIT 20""").fetchall()
if not rows:
    print('  No leader events for profileId=live')
else:
    for r in rows:
        print(f'  {fmt_ts_ms(r["ts"])} {r["side"]:4} price={r["price"]:.4f} usdc={r["usdcSize"]:.2f}  token=...{r["tokenId"][-12:]}')

# 6. Check ALL intents skip reasons summary
print()
print('=== LIVE PROFILE - Skip reason summary ===')
rows = db.execute("""SELECT reason, COUNT(*) as cnt FROM CopyIntent
                     WHERE profileId='live' GROUP BY reason ORDER BY cnt DESC""").fetchall()
if not rows:
    print('  No intents at all for profileId=live')
else:
    total = sum(r["cnt"] for r in rows)
    for r in rows:
        pct = 100 * r["cnt"] / total
        print(f'  {r["reason"] or "FILLED":35} {r["cnt"]:5} ({pct:.1f}%)')

# 7. Check bot.feed.last_error metric for any profile
print()
print('=== FEED ERRORS (any profile) ===')
rows = db.execute("""SELECT profileId, key, value, updatedAt FROM RuntimeMetric
                     WHERE key LIKE '%error%' OR key LIKE '%feed%'""").fetchall()
if not rows:
    print('  No feed error metrics found')
else:
    for r in rows:
        print(f'  [{r["profileId"]}] {r["key"]}: {r["value"][:120]}')

db.close()

File: scripts/diagnose-live.mjs
import Database from 'better-sqlite3';
import { resolve } from 'path';
import { fileURLToPath } from 'url';
import { dirname } from 'path';

const __dirname = dirname(fileURLToPath(import.meta.url));
const dbPath = resolve(__dirname, '../packages/db/prisma/dev.db');
const db = new Database(dbPath, { readonly: true });

// Kill switch state
console.log('=== CONFIG / KILL SWITCH ===');
const cfg = db.prepare("SELECT json FROM ConfigVersion WHERE active=1 ORDER BY createdAt DESC LIMIT 1").get();
if (cfg) {
  const c = JSON.parse(cfg.json);
  console.log('  killSwitch:', c.safety?.killSwitch, '  paused:', c.safety?.paused);
} else {
  console.log('  no active config');
}

// Recent errors
console.log('\n=== RECENT LIVE_EXECUTION_ERRORs (last 10) ===');
const errs = db.prepare("SELECT ts, contextJson FROM Alert WHERE code='LIVE_EXECUTION_ERROR' ORDER BY ts DESC LIMIT 10").all();
if (errs.length === 0) console.log('  none');
errs.forEach(r => {
  const meta = JSON.parse(r.contextJson);
  console.log(' ', new Date(r.ts).toLocaleTimeString(), meta.error?.slice(0, 140));
});

// Recent error storms
console.log('\n=== ERROR STORMS (last 5) ===');
const storms = db.prepare("SELECT ts, message FROM Alert WHERE code='ERROR_STORM' ORDER BY ts DESC LIMIT 5").all();
if (storms.length === 0) console.log('  none');
storms.forEach(r => console.log(' ', new Date(r.ts).toLocaleTimeString(), r.message));

// Intent summary
console.log('\n=== INTENT TOTALS ===');
db.prepare("SELECT status, reason, COUNT(*) as n FROM CopyIntent GROUP BY status, reason ORDER BY n DESC LIMIT 20").all()
  .forEach(r => console.log(' ', String(r.n).padStart(6), r.status, r.reason ?? ''));

// Fills
console.log('\n=== RECENT FILLS (last 20) ===');
const fills = db.prepare(`
  SELECT ci.ts, ci.side, ci.tokenId, f.price, f.size
  FROM CopyIntent ci
  JOIN CopyOrder co ON co.intentId = ci.id
  JOIN Fill f ON f.orderId = co.id
  ORDER BY ci.ts DESC LIMIT 20
`).all();
if (fills.length === 0) console.log('  none');
fills.forEach(r => {
  const notional = (r.price * r.size).toFixed(2);
  console.log(' ', new Date(r.ts).toLocaleTimeString(), r.side.padEnd(4), 'price:', String(r.price).padEnd(6), 'size:', String(r.size).padEnd(8), 'USDC:', notional);
});

// Total spend vs position value estimate
const totalBuyNotional = db.prepare("SELECT COALESCE(SUM(f.price * f.size), 0) as total FROM CopyIntent ci JOIN CopyOrder co ON co.intentId=ci.id JOIN Fill f ON f.orderId=co.id WHERE ci.side='BUY'").get();
const totalSellNotional = db.prepare("SELECT COALESCE(SUM(f.price * f.size), 0) as total FROM CopyIntent ci JOIN CopyOrder co ON co.intentId=ci.id JOIN Fill f ON f.orderId=co.id WHERE ci.side='SELL'").get();
console.log('\n=== SPEND SUMMARY ===');
console.log('  Total BUY  USDC spent:    $' + Number(totalBuyNotional.total).toFixed(2));
console.log('  Total SELL USDC received: $' + Number(totalSellNotional.total).toFixed(2));
console.log('  Net USDC out (open risk):  $' + (totalBuyNotional.total - totalSellNotional.total).toFixed(2));

db.close();
